app/models/instances.rb

```diff
@@ -0,0 +1,96 @@
+class Instances < Base
+  def list(service_id, cert)
+    occi_client(endpoint(service_id), cert)
+    parse_instances(@client.describe(Occi::Infrastructure::Compute.new.kind.type_identifier))
+  end
+
+  def show(service_id, cert, instance_id)
+    occi_client(endpoint(service_id), cert)
+    instance_url = validate(service_id, cert, instance_id)
+    parse_instances(@client.describe(instance_url))
+  end
+
+  def destroy(service_id, cert, instance_id)
+    occi_client(endpoint(service_id), cert)
+    instance_url = validate(service_id, cert, instance_id)
+    @client.delete instance_url
+  end
+
+  def create(service_id, cert, instance_id)
+    # occi_client(endpoint(service_id), cert)
+    # instance_url = validate(service_id, cert, instance_id)
+    # @client.delete instance_url
+  end
+
+  # TODO: remove when released
+  def all_test_method(service_id, cert)
+    occi_client(endpoint(service_id), cert)
+    @client.describe Occi::Infrastructure::Compute.new.kind.type_identifier
+  end
+
+  protected
+
+  def validate(service_id, _cert, instance_id)
+    instance_url = endpoint(service_id) + Base64.decode64(instance_id)
+    return instance_url if (@client.list Occi::Infrastructure::Compute.new.kind.type_identifier).include? instance_url
+    raise Errors::NotFoundError, "Instance '#{instance_id}' at site '#{service_id}' could not be found!"
+  end
+
+  def parse_instances(computes)
+    computes.collect do |cmpt|
+      {
+        id: Base64.strict_encode64(cmpt.location),
+        name: (cmpt.title || cmpt.hostname),
+        # credentials: parse_credentials(cmpt),
+        applianceId: appliance_id(cmpt),
+        flavourId: flavour_mixin(cmpt).to_s,
+        userData: parse_user_data(cmpt),
+        architecture: cmpt.architecture,
+        state: cmpt.state
+      }
+    end
+  end
+
+  def appliance_id(compute)
+    appliance = appliance_mixin(compute)
+    appliance.term if appliance.present?
+  end
+
+  def appliance_mixin(compute)
+    model = @client.model
+    compute.mixins.find do |mixin|
+      # Awfull workaround due to bug in occi core, should be fixed in new occi client with
+      # mixin.related_to? (Occi::Infrastructure::OsTpl.mixin.type_identifier)
+      model.get_by_id(mixin.type_identifier).present? && \
+        model.get_by_id(mixin.type_identifier).related_to?(Occi::Infrastructure::OsTpl.mixin.type_identifier)
+    end
+  end
+
+  def flavour_mixin(compute)
+    compute.mixins.find { |mixin| mixin.scheme == 'http://fedcloud.egi.eu/occi/compute/flavour/1.0#' }
```

Comment:
I would say this is not correct. I think this should look similar to the `appliance_mixin` method, shouldn't it?